lib/ansible/modules/system/listen_ports_facts.py

```
@@ -0,0 +1,312 @@
+    return processes
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            whitelist_tcp = dict(required=False, type='list', default=list()),
+            whitelist_udp = dict(required=False, type='list', default=list())
+        ),
+        supports_check_mode=False
+    )
+
+    if platform.system() != 'Linux':
+        module.fail_json(msg='This module requires Linux.')
+
+    def getPidSTime(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'lstart', '-p', str(pid)])
+        stime = ''
+        if rc == 0:
+            for line in iter(ps_output.splitlines()):
+                if 'started' not in line:
+                    stime = line
+        return stime
+
+    def getPidUser(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'user', '-p', str(pid)])
+        user = ''
+        if rc == 0:
+            for line in iter(ps_output.splitlines()):
+                if line != 'USER':
+                    user = line
+        return user
+
+    result = {}
+    result['changed'] = False
+    result['ansible_facts'] = dict(tcp_listen_violations=list(), udp_listen_violations=list(), tcp_listen=list(), udp_listen=list())
+
+    try:
+        netstat_cmd = module.get_bin_path('netstat', True)
+
+        # which ports are listening for connections?
+        rc, stdout, stderr = module.run_command([netstat_cmd, '-plunt'])
+        if rc == 0:
+            netstatOut = netStatParse(stdout)
+            for i, p in enumerate(netstatOut):
+                p['stime'] = getPidSTime(p['pid'])
+                p['user'] = getPidUser(p['pid'])
+                if p['protocol'] == 'tcp':
+                    result['ansible_facts']['tcp_listen'].append(p)
+                elif p['protocol'] == 'udp':
+                    result['ansible_facts']['udp_listen'].append(p)
+
+    except EnvironmentError:
```

I would catch all exceptions here, so that the output is clean JSON.
Please use this common code:
```python
    except:
        e = get_exception()
        module.fail_json(msg=str(e))
```